core/src/main/java/org/apache/accumulo/core/security/ColumnVisibility.java

```diff
@@ -125,6 +125,23 @@ public Node(int start, int end) {
       this.end = end;
     }
 
+    /**
+     * Creates a new node by performing a deep copy of an existing node object
+     *
+     * @param node Node object
+     * @since 2.1.4
+     */
+    public Node(Node node) {
+      List<Node> childrenNew = new ArrayList<>(node.children.size());
+      for (Node child : node.children) {
+        childrenNew.add(new Node(child));
+      }
```

Comment:
Looked over the existing code and noticed it had some handling to avoid allocating empty list in the nodes. Can do that here also.

```suggestion
      List<Node> childrenNew = node.children.isEmpty() ? EMPTY : new ArrayList<>(node.children.size());
      for (Node child : node.children) {
        childrenNew.add(new Node(child));
      }
```

Reply:
Sounds good